### ISSUE TITLE
meshctl cluster register error apiGroup has invalid type: got string, want pointer

### ISSUE THREAD

**Issue by rvennam** [solo-io/gloo-mesh#1225]
**Describe the bug**
```
 $ meshctl cluster register   --cluster-name mgmt-cluster   --remote-context $MGMT_CONTEXT
INFO[0005] finished upgrading chart agent-crds as release
INFO[0009] finished upgrading chart cert-agent as release
panic: field k8s.io.api.rbac.v1.RoleRef.apiGroup has invalid type: got string, want pointer

goroutine 1 [running]:
google.golang.org/protobuf/internal/impl.fieldInfoForScalar(0x384e180, 0xc000844580, 0x30fefa8, 0x8, 0x0, 0x0, 0x384e2a0, 0x30fe080, 0x30fefb2, 0x34, ...)
	/go/pkg/mod/google.golang.org/protobuf@v1.25.0/internal/impl/message_reflect_field.go:228 +0xa58
google.golang.org/protobuf/internal/impl.(*MessageInfo).makeKnownFieldsFunc(0xc000294280, 0xffffffffffffffff, 0xffffffffffffffff, 0xffffffffffffffff, 0xffffffffffffffff, 0xc001cc7f50, 0xc001cc7f80, 0xc001cc7fb0, 0xc001a86000)
	/go/pkg/mod/google.golang.org/protobuf@v1.25.0/internal/impl/message_reflect.go:67 +0xbd4
google.golang.org/protobuf/internal/impl.(*MessageInfo).makeReflectFuncs(0xc000294280, 0x384e2a0, 0x32aee60, 0xffffffffffffffff, 0xffffffffffffffff, 0xffffffffffffffff, 0xffffffffffffffff, 0xc001cc7f50, 0xc001cc7f80, 0xc001cc7fb0, ...)
	/go/pkg/mod/google.golang.org/protobuf@v1.25.0/internal/impl/message_reflect.go:36 +0x63
google.golang.org/protobuf/internal/impl.(*MessageInfo).initOnce(0xc000294280)
	/go/pkg/mod/google.golang.org/protobuf@v1.25.0/internal/impl/message.go:90 +0x2f1
google.golang.org/protobuf/internal/impl.(*MessageInfo).init(0xc000294280)
	/go/pkg/mod/google.golang.org/protobuf@v1.25.0/internal/impl/message.go:72 +0x3f
google.golang.org/protobuf/internal/impl.(*messageReflectWrapper).Range(0xc000655fe0, 0xc001cc7f20)
	/go/pkg/mod/google.golang.org/protobuf@v1.25.0/internal/impl/message_reflect_gen.go:164 +0x56
google.golang.org/protobuf/proto.equalMessage(0x38417c0, 0xc000655fe0, 0x38417c0, 0xc000655ff0, 0x3392d00)
	/go/pkg/mod/google.golang.org/protobuf@v1.25.0/proto/equal.go:52 +0x1ee
google.golang.org/protobuf/proto.Equal(0x37c10a0, 0xc000655fe0, 0x37c10a0, 0xc000655ff0, 0x3392d00)
	/go/pkg/mod/google.golang.org/protobuf@v1.25.0/proto/equal.go:41 +0x161
github.com/golang/protobuf/proto.Equal(0x8abc6f0, 0xc001cc7e30, 0x8abc6f0, 0xc001cc7e90, 0xc001cc7e00)
	/go/pkg/mod/github.com/golang/protobuf@v1.4.3/proto/proto.go:161 +0x120
github.com/solo-io/skv2/pkg/equalityutils.DeepEqual(0x3392d60, 0xc001cc7e30, 0x3392d60, 0xc001cc7e90, 0xc001cc7e00)
	/go/pkg/mod/github.com/solo-io/skv2@v0.17.0/pkg/equalityutils/deep_equal.go:18 +0x13f
github.com/solo-io/skv2/pkg/controllerutils.ObjectsEqual(0x37d3fe0, 0xc0002d9a20, 0x37d3fe0, 0xc0002d98c0, 0xc0002d9800)
	/go/pkg/mod/github.com/solo-io/skv2@v0.17.0/pkg/controllerutils/equality.go:56 +0xc69
github.com/solo-io/skv2/pkg/controllerutils.Upsert(0x380c4e0, 0xc0000d0000, 0x38331e0, 0xc00063c070, 0x38523e0, 0xc0002d98c0, 0xc00148c918, 0x1, 0x1, 0x0, ...)
	/go/pkg/mod/github.com/solo-io/skv2@v0.17.0/pkg/controllerutils/upsert.go:44 +0x516
github.com/solo-io/skv2/pkg/multicluster/internal/k8s/rbac.authorization.k8s.io/v1.(*clusterRoleBindingClient).UpsertClusterRoleBinding(0xc000783190, 0x380c4e0, 0xc0000d0000, 0xc0002d98c0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/pkg/mod/github.com/solo-io/skv2@v0.17.0/pkg/multicluster/internal/k8s/rbac.authorization.k8s.io/v1/clients.go:633 +0x194
github.com/solo-io/skv2/pkg/multicluster/register/internal.(*clusterRbacBinder).bindClusterRolesToServiceAccount(0xc000bb1ce0, 0x380c4e0, 0xc0000d0000, 0xc004b47220, 0x9, 0xc004b47210, 0xc, 0xc000bb1d00, 0x1, 0x1, ...)
	/go/pkg/mod/github.com/solo-io/skv2@v0.17.0/pkg/multicluster/register/internal/cluster_rbac_binder.go:92 +0x3d7
github.com/solo-io/skv2/pkg/multicluster/register/internal.(*clusterRbacBinder).BindClusterRoles(0xc000bb1ce0, 0x380c4e0, 0xc0000d0000, 0xc004b47220, 0x9, 0xc004b47210, 0xc, 0xc000bb1d00, 0x1, 0x1, ...)
	/go/pkg/mod/github.com/solo-io/skv2@v0.17.0/pkg/multicluster/register/internal/cluster_rbac_binder.go:66 +0xf0
github.com/solo-io/skv2/pkg/multicluster/register.(*clusterRegistrant).CreateRemoteAccessToken(0xc000207e00, 0x380c4e0, 0xc0000d0000, 0x380cfa0, 0xc000139270, 0xc004b47220, 0x9, 0xc004b47210, 0xc, 0x7ffeefbff6f0, ...)
	/go/pkg/mod/github.com/solo-io/skv2@v0.17.0/pkg/multicluster/register/registrant.go:236 +0xe1e
github.com/solo-io/skv2/pkg/multicluster/register.RegisterClusterFromConfig(0x380c4e0, 0xc0000d0000, 0xc00060b440, 0x380cfa0, 0xc000139270, 0x7ffeefbff6f0, 0xc, 0x34418fe, 0x9, 0x7ffeefbff70e, ...)
	/go/pkg/mod/github.com/solo-io/skv2@v0.17.0/pkg/multicluster/register/helpers.go:232 +0x3a5
github.com/solo-io/skv2/pkg/multicluster/register.RegistrationOptions.RegisterProviderCluster(0x380cfa0, 0xc0001391d0, 0x380cfa0, 0xc000139270, 0x7ffeefbff70e, 0x25, 0x0, 0x0, 0x7ffeefbff6f0, 0xc, ...)
	/go/pkg/mod/github.com/solo-io/skv2@v0.17.0/pkg/multicluster/register/helpers.go:110 +0x2e0
github.com/solo-io/skv2/pkg/multicluster/register.RegistrationOptions.RegisterCluster(0x380cfa0, 0xc0001391d0, 0x380cfa0, 0xc000139270, 0x7ffeefbff70e, 0x25, 0x0, 0x0, 0x7ffeefbff6f0, 0xc, ...)
	/go/pkg/mod/github.com/solo-io/skv2@v0.17.0/pkg/multicluster/register/helpers.go:93 +0x8c
github.com/solo-io/gloo-mesh/pkg/meshctl/registration.(*Registrant).registerCluster(0xc000371f68, 0x380c4e0, 0xc0000d0000, 0x0, 0x0)
	/workspace/gloo-mesh/pkg/meshctl/registration/register.go:139 +0x19c
github.com/solo-io/gloo-mesh/pkg/meshctl/registration.(*Registrant).RegisterCluster(0xc000371f68, 0x380c4e0, 0xc0000d0000, 0x0, 0x0)
	/workspace/gloo-mesh/pkg/meshctl/registration/register.go:117 +0x356
github.com/solo-io/gloo-mesh/pkg/meshctl/commands/cluster/register.Command.func1(0xc000128dc0, 0xc0000bae80, 0x0, 0x4, 0x0, 0x0)
	/workspace/gloo-mesh/pkg/meshctl/commands/cluster/register/cluster_register.go:25 +0x17f
github.com/spf13/cobra.(*Command).execute(0xc000128dc0, 0xc0000bae40, 0x4, 0x4, 0x0, 0x0)
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:842 +0xb21
github.com/spf13/cobra.(*Command).ExecuteC(0xc000128580, 0xc000128dc0, 0x0, 0x0)
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x70f
github.com/spf13/cobra.(*Command).Execute(0xc000128580, 0x0, 0x0)
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887 +0x3b
main.main()
	/workspace/gloo-mesh/cmd/meshctl/main.go:11 +0x5a
```
**To Reproduce**
1. Installed gloo mesh on mgmt-cluster
2. Installed istio on mgmt and remote clusters
3. Tried to register mgmt cluster to gloo mesh

**Additional context**
Add any other context about the problem here, e.g.
- Gloo Mesh version "0.12.2"
- Kubernetes version 1.18
- service mesh and version (if applicable, e.g. Istio 1.7) Istio 1.8

**Comment by joekelley** [comment]
Thanks for filing this. I was able to reproduce by re-registering a cluster that has already been registered. It appears that there's a bug in the equality function used by our Upsert function to create-or-update the ClusterRoleBinding used on the remote cluster.

**Comment by joekelley** [comment]
More details in https://github.com/solo-io/skv2/issues/207

**Comment by kdorosh** [comment]
fix is in skv2 0.17.2

**Comment by kdorosh** [comment]
reopening until gloo-mesh picks up the dep bump
